gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/connectedcomponents/ValueWeaklyConnectedComponents.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.connectedcomponents;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.utils.DataSetUtils;
+import org.apache.flink.graph.Edge;
+import org.apache.flink.graph.Graph;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.graph.library.ConnectedComponents;
+import org.apache.flink.types.NullValue;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.flink.algorithms.gelly.BaseGellyAlgorithm;
+import org.gradoop.flink.algorithms.gelly.connectedcomponents.functions.CreateLongSourceIds;
+import org.gradoop.flink.algorithms.gelly.connectedcomponents.functions.CreateLongTargetIds;
+import org.gradoop.flink.algorithms.gelly.functions.LongTupleToGellyEdgeWithLongValue;
+import org.gradoop.flink.algorithms.gelly.functions.LongTupleToGellyVertexWithLongValue;
+import org.gradoop.flink.algorithms.gelly.connectedcomponents.functions.MapVertexIdComponentId;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.model.impl.functions.epgm.SourceId;
+
+/**
+ * A gradoop operator wrapping Flinks ScatterGatherIteration-Algorithm for ConnectedComponents
+ * {@link org.apache.flink.graph.library.ConnectedComponents}.
+ */
```

Comment:
Please use a more specific description here, e.g. a short detailing of the return value like "...returns a mapping VertexId -> ComponentId..."